diff --git a/Mods/aircraft/A-10C/Cockpit/Scripts/AN_ALR69V/indicator/AN_ALR69V_init.lua b/Mods/aircraft/A-10C/Cockpit/Scripts/AN_ALR69V/indicator/AN_ALR69V_init.lua
index 20b5e79..9c5cb3b 100644
--- a/Mods/aircraft/A-10C/Cockpit/Scripts/AN_ALR69V/indicator/AN_ALR69V_init.lua
+++ b/Mods/aircraft/A-10C/Cockpit/Scripts/AN_ALR69V/indicator/AN_ALR69V_init.lua
@@ -1,6 +1,7 @@
 dofile(LockOn_Options.common_script_path.."devices_defs.lua")
 
 indicator_type = indicator_types.COMMON
+purposes		= {render_purpose.GENERAL}
 ----------------------
 rwr_symbols_bound = 0.027   -- coeff to limit indication on RWR when symbol can't be place on RWR entirely
 -------PAGE IDs-------
@@ -56,3 +57,5 @@ opacity_sensitive_materials =
 "INDICATION_RWR"
 }
 
+dofile(LockOn_Options.common_script_path.."ViewportHandling.lua")
+try_find_assigned_viewport("A10C_RWR", "RWR_SCREEN")
diff --git a/Mods/aircraft/A-10C/Cockpit/Scripts/CDU/indicator/CDU_init.lua b/Mods/aircraft/A-10C/Cockpit/Scripts/CDU/indicator/CDU_init.lua
index 41ef60c..c6ba37e 100644
--- a/Mods/aircraft/A-10C/Cockpit/Scripts/CDU/indicator/CDU_init.lua
+++ b/Mods/aircraft/A-10C/Cockpit/Scripts/CDU/indicator/CDU_init.lua
@@ -1,6 +1,7 @@
 dofile(LockOn_Options.common_script_path.."devices_defs.lua")
 
 indicator_type = indicator_types.COMMON
+purposes		= {render_purpose.GENERAL}
 
 dofile(LockOn_Options.script_path.."CDU/indicator/CDU_pages.lua")
 init_pageID	= id_CDU_Pages.CDU_PAGE_OFF
@@ -28,5 +29,5 @@ dofile(LockOn_Options.common_script_path.."devices_defs.lua")
 indicator_type = indicator_types.COMMON
 used_render_mask = "interleave.dds"
 
-
-
+dofile(LockOn_Options.common_script_path.."ViewportHandling.lua")
+try_find_assigned_viewport("A10C_CDU", "CDU_SCREEN")
diff --git a/Mods/aircraft/A-10C/Cockpit/Scripts/CMSC/indicator/CMSC_init.lua b/Mods/aircraft/A-10C/Cockpit/Scripts/CMSC/indicator/CMSC_init.lua
index 8b9a3b2..abb23b9 100644
--- a/Mods/aircraft/A-10C/Cockpit/Scripts/CMSC/indicator/CMSC_init.lua
+++ b/Mods/aircraft/A-10C/Cockpit/Scripts/CMSC/indicator/CMSC_init.lua
@@ -30,3 +30,5 @@ pages[id_Page.PAGE_MAIN] = {id_pagesubset.MAIN}
 init_pageID     = id_Page.PAGE_MAIN
 purposes 	 = {render_purpose.GENERAL}
 
+dofile(LockOn_Options.common_script_path.."ViewportHandling.lua")
+try_find_assigned_viewport("A10C_CMSC", "CMSC_SCREEN")
diff --git a/Mods/aircraft/A-10C/Cockpit/Scripts/CMSP/indicator/CMSP_init.lua b/Mods/aircraft/A-10C/Cockpit/Scripts/CMSP/indicator/CMSP_init.lua
index 2ea4615..fd9df6b 100644
--- a/Mods/aircraft/A-10C/Cockpit/Scripts/CMSP/indicator/CMSP_init.lua
+++ b/Mods/aircraft/A-10C/Cockpit/Scripts/CMSP/indicator/CMSP_init.lua
@@ -30,3 +30,5 @@ pages[id_Page.PAGE_MAIN] = {id_pagesubset.MAIN}
 init_pageID     = id_Page.PAGE_MAIN
 purposes 	 = {render_purpose.GENERAL}
 
+dofile(LockOn_Options.common_script_path.."ViewportHandling.lua")
+try_find_assigned_viewport("A10C_CMSP", "CMSP_SCREEN")
diff --git a/Mods/aircraft/A-10C/Cockpit/Scripts/DigitalClock/Indicator/DIGIT_CLK_init.lua b/Mods/aircraft/A-10C/Cockpit/Scripts/DigitalClock/Indicator/DIGIT_CLK_init.lua
index 28e906d..257b6fa 100644
--- a/Mods/aircraft/A-10C/Cockpit/Scripts/DigitalClock/Indicator/DIGIT_CLK_init.lua
+++ b/Mods/aircraft/A-10C/Cockpit/Scripts/DigitalClock/Indicator/DIGIT_CLK_init.lua
@@ -1,6 +1,7 @@
 dofile(LockOn_Options.common_script_path.."devices_defs.lua")
 
 indicator_type = indicator_types.COMMON
+purposes		= {render_purpose.GENERAL}
 
 -------PAGE IDs-------
 id_Page =
@@ -40,3 +41,7 @@ end
 
 pages_by_mode[A10C_DIGIT_CLK_OFF][0][0][0]			  = id_Page.PAGE_OFF
 pages_by_mode[A10C_DIGIT_CLK_MAIN][0][0][0]			  = id_Page.PAGE_MAIN
+
+
+dofile(LockOn_Options.common_script_path.."ViewportHandling.lua")
+try_find_assigned_viewport("A10C_CLOCK", "DIGIT_CLOCK")
\ No newline at end of file
diff --git a/Mods/aircraft/A-10C/Cockpit/Scripts/UHF_Radio/indicator/freq_status_init.lua b/Mods/aircraft/A-10C/Cockpit/Scripts/UHF_Radio/indicator/freq_status_init.lua
index 108aaff..eb42342 100644
--- a/Mods/aircraft/A-10C/Cockpit/Scripts/UHF_Radio/indicator/freq_status_init.lua
+++ b/Mods/aircraft/A-10C/Cockpit/Scripts/UHF_Radio/indicator/freq_status_init.lua
@@ -1,4 +1,9 @@
 dofile(LockOn_Options.script_path.."UHF_Radio/indicator/common_init.lua")
 
+purposes = {render_purpose.GENERAL}
+
 dofile(LockOn_Options.script_path.."UHF_Radio/indicator/indicators.lua")
 indicator = indicators.FREQUENCY_STATUS
+
+dofile(LockOn_Options.common_script_path.."ViewportHandling.lua")
+try_find_assigned_viewport("A10C_UHFF", "UHF_FREQUENCY_STATUS")
diff --git a/Mods/aircraft/A-10C/Cockpit/Scripts/UHF_Radio/indicator/preset_channel_init.lua b/Mods/aircraft/A-10C/Cockpit/Scripts/UHF_Radio/indicator/preset_channel_init.lua
index edfaf9c..0625fa9 100644
--- a/Mods/aircraft/A-10C/Cockpit/Scripts/UHF_Radio/indicator/preset_channel_init.lua
+++ b/Mods/aircraft/A-10C/Cockpit/Scripts/UHF_Radio/indicator/preset_channel_init.lua
@@ -1,4 +1,9 @@
 dofile(LockOn_Options.script_path.."UHF_Radio/indicator/common_init.lua")
 
+purposes = {render_purpose.GENERAL}
+
 dofile(LockOn_Options.script_path.."UHF_Radio/indicator/indicators.lua")
 indicator = indicators.PRESET_CHANNEL
+
+dofile(LockOn_Options.common_script_path.."ViewportHandling.lua")
+try_find_assigned_viewport("A10C_UHFP", "UHF_PRESET_CHANNEL")
diff --git a/Mods/aircraft/A-10C/Cockpit/Scripts/UHF_Radio/indicator/repeater_init.lua b/Mods/aircraft/A-10C/Cockpit/Scripts/UHF_Radio/indicator/repeater_init.lua
index 19c6de4..0f38510 100644
--- a/Mods/aircraft/A-10C/Cockpit/Scripts/UHF_Radio/indicator/repeater_init.lua
+++ b/Mods/aircraft/A-10C/Cockpit/Scripts/UHF_Radio/indicator/repeater_init.lua
@@ -1,4 +1,9 @@
 dofile(LockOn_Options.script_path.."UHF_Radio/indicator/common_init.lua")
 
+purposes = {render_purpose.GENERAL}
+
 dofile(LockOn_Options.script_path.."UHF_Radio/indicator/indicators.lua")
 indicator = indicators.REPEATER
+
+dofile(LockOn_Options.common_script_path.."ViewportHandling.lua")
+try_find_assigned_viewport("A10C_UHFR", "UHF_REPEATER")
